pandas/compat/numpy/function.py

```diff
@@ -245,3 +253,48 @@ def validate_transpose_for_generic(inst, kwargs):
             msg += " for {klass} instances".format(klass=klass)
 
         raise ValueError(msg)
+
+
+def validate_window_func(name, args, kwargs):
+    numpy_args = ('axis', 'dtype', 'out')
+    msg = ("numpy operations are not "
+           "valid with window objects. "
+           "Call the object's '{func}' "
+           "method directly instead".format(func=name))
```

Comment:
I guess the best way to do it would be to write different validation methods for each type (`window`, `rolling`, `expanding`, `ewm`).  That way we can get more distinct validation error messages.